app/src/main/java/com/asfoundation/wallet/logging/RakamReceiver.kt

```diff
@@ -0,0 +1,48 @@
+package com.asfoundation.wallet.logging
+
+import com.asfoundation.wallet.logging.LogReceiver.Companion.DEFAULT_MSG
+import com.asfoundation.wallet.logging.LogReceiver.Companion.DEFAULT_TAG
+import com.asfoundation.wallet.logging.LogReceiver.Companion.DEFAULT_THROWABLE_STATCKTRACE
+import io.rakam.api.Rakam
+import org.json.JSONException
+import org.json.JSONObject
+
+class RakamReceiver : LogReceiver {
+  companion object {
+    private const val LOG_EVENT_TYPE = "wallet_non_fatal_event"
+  }
+
+  override fun log(tag: String?, throwable: Throwable?) {
+    Rakam.getInstance()
+        .logEvent(LOG_EVENT_TYPE, map(tag = tag, throwable = throwable))
+  }
+
+  override fun log(tag: String?, message: String?) {
+    Rakam.getInstance()
+        .logEvent(LOG_EVENT_TYPE, map(tag = tag, message = message))
+  }
+
+  override fun log(tag: String?, message: String?, throwable: Throwable?) {
+    Rakam.getInstance()
+        .logEvent(LOG_EVENT_TYPE, map(tag, message, throwable))
+  }
+
+  private fun map(tag: String? = DEFAULT_TAG, message: String? = DEFAULT_MSG,
+                  throwable: Throwable? = Throwable()): JSONObject {
```

Comment:
Do we want to send the throwable_message property if there's no throwable? I don't think the default values are needed here

Reply:
Agreed